src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/ClientInterface.cs

```diff
@@ -0,0 +1,453 @@
+        // adjacent disallowed characters.
+        //
+        // Permitted characters are:
+        //
+        //   - ASCII letters ([A-za-z])
+        //   - ASCII digits ([0-9])
+        //   - Space (' ')
+        //   - Period ('.')
+        //   - Plus ('+')
+        //   - Underscore ('_')
+        //   - Hyphen ('-')
+        //
+        // If the given value is null, empty, or all whitespace, or an error
+        // occurs, the fallback value is returned.
+        //
+        public static string Clean(string? value)
+        {
+            if (string.IsNullOrWhiteSpace(value)
+#if NETFRAMEWORK
+                // .NET Framework doesn't consider IsNullOrWhiteSpace()
+                // sufficient for nullable checks, so add an explicit check for
+                // null.
+                || value == null
+#endif // NETFRAMEWORK
+               )
+            {
+                return Unknown;
+            }
+            
+            // Remove any leading and trailing whitespace.
+            value = value.Trim();
+
+            try
+            {
+                // Build the cleaned value by hand, avoiding the overhead and
+                // failure scenarios of regexes or other more complex solutions.
+                //
+                // We expect the value to be short, and this code is called only
+                // a few times per process.  Robustness and simplicity are more
+                // important than performance here.
+                //
+                StringBuilder cleaned = new StringBuilder(value.Length);
+                foreach (char c in value)
+                {
+                    // Is it a permitted character?
+                    if (
+#if NET
+                        char.IsAsciiLetter(c)
+                        || char.IsAsciiDigit(c)
+#else
+                        (c >= 'A' && c <= 'Z')
+                        || (c >= 'a' && c <= 'z')
+                        || (c >= '0' && c <= '9')
+#endif // NET
+                        || c == ' '
+                        || c == '.'
+                        || c == '+'
+                        || c == '_'
+                        || c == '-')
+                    {
+                        // Yes, so append it as-is.
+                        cleaned.Append(c);
+                    }
+                    else
+                    {
+                        // No, so replace it with an underscore.
+                        cleaned.Append('_');
+                    }
+                }
+
+                return cleaned.ToString();
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                // StringBuilder failed in an unexpected way, so use our
+                // fallback value.
+                return Unknown;
+            }
+        }
+
+        // Truncate the given value to the given max length, and return the
+        // result.
+        public static string Trunc(string value, ushort maxLength)
```

Comment:
Will you consider renaming this to `Truncate`. 
Also xmldocs style comment would be nice